### PR TITLE
Don't feed remaining subtitles when audio skips to past the end of last subtitle

### PR DESCRIPTION
**Pull request by ZarothYe**
In the Yeesha Cleft speech, it appears that when you press the imager button to stop the hologram in the middle of her speech, the audio skips close to the ending position and stays there for several `Update` cycles before actually ending, which was causing the updater to feed all the remaining subtitles up to that point. This doesn't seem to be the case when Kirel podium audio is stopped, so I don't exactly know why the Yeesha audio does this. Added a check to make sure the current position/time isn't past the end of the last subtitle before requesting to display entries up to that time.

This will collide with #1096 , so please let me rebase and merge/fix before merging this one.

### CODE REVIEW

File: Sources/Plasma/PubUtilLib/plAudioCore/plSrtFileReader.h [hunk 1]
```diff
@@ -92,6 +92,7 @@ class plSrtFileReader
     void            AdvanceToTime(uint32_t timeMs);
     plSrtEntry*     GetNextEntryStartingBeforeTime(uint32_t timeMs);
     plSrtEntry*     GetNextEntryEndingBeforeTime(uint32_t timeMs);
+    uint32_t        GetLastEntryEndTime();
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAudioCore/plSrtFileReader.h:95]:
```suggestion
    uint32_t        GetLastEntryEndTime() const;
```

File: Sources/Plasma/PubUtilLib/plAudioCore/plSrtFileReader.cpp [hunk 2]
```diff
@@ -181,3 +181,12 @@ plSrtEntry* plSrtFileReader::GetNextEntryEndingBeforeTime(uint32_t timeMs)
 
     return nullptr;
 }
+
+uint32_t plSrtFileReader::GetLastEntryEndTime()
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAudioCore/plSrtFileReader.cpp:185]:
```suggestion
uint32_t plSrtFileReader::GetLastEntryEndTime() const
```

File: Sources/Plasma/PubUtilLib/plAudio/plWin32Sound.cpp [hunk 3]
```diff
@@ -106,11 +106,14 @@ void plWin32Sound::Update()
...
         plSrtFileReader* srtReader = buf->GetSrtReader();
         if (srtReader != nullptr) {
-            while (plSrtEntry* nextEntry = srtReader->GetNextEntryStartingBeforeTime((uint32_t)(GetActualTimeSec() * 1000.0f))) {
-                if (plgAudioSys::AreSubtitlesEnabled()) {
-                    // add a plSubtitleMsg to go... to whoever is listening (probably the KI)
-                    plSubtitleMsg* msg = new plSubtitleMsg(nextEntry->GetSubtitleText(), nextEntry->GetSpeakerName());
-                    msg->Send();
+            uint32_t currentTimeMs = (uint32_t)(GetActualTimeSec() * 1000.0f);
+            if (currentTimeMs <= srtReader->GetLastEntryEndTime()) {
+                while (plSrtEntry* nextEntry = srtReader->GetNextEntryStartingBeforeTime((uint32_t)(GetActualTimeSec() * 1000.0f))) {
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAudio/plWin32Sound.cpp:111]: Is `(uint32_t)(GetActualTimeSec() * 1000.0f)` the same value as `currentTime` here?

Comment (ZarothYe) [reply Sources/Plasma/PubUtilLib/plAudio/plWin32Sound.cpp:111]: Well dangit, I think I borked that in the merge. Don't do merges late at night, kids. Yes, the `currentTime` variable was supposed to be used in both places. I'll fix.

**Review by Hoikas — APPROVED** [review]

**Pull request merged**